sql/core_clients_last_seen_raw_v1.sql

```diff
@@ -0,0 +1,86 @@
+  -- Equivalent to, but more efficient than, calling udf_bitmask_range(1, 28)
+CREATE TEMP FUNCTION
+  udf_bitmask_lowest_28() AS (0x0FFFFFFF);
+  --
+CREATE TEMP FUNCTION
+  udf_shift_one_day(x INT64) AS (IFNULL((x << 1) & udf_bitmask_lowest_28(),
+      0));
+  --
+CREATE TEMP FUNCTION
+  udf_combine_days(prev INT64,
+    curr INT64) AS (udf_shift_one_day(prev) + IFNULL(curr,
+      0));
+  --
+WITH
+  _current AS (
+  SELECT
+    -- In this raw table, we capture the history of activity over the past
+    -- 28 days for each usage criterion as a single 64-bit integer. The
+    -- rightmost bit represents whether the user was active in the current day.
+    CAST(TRUE AS INT64) AS days_seen_bits,
```

Comment:
@jmccrosky - Do you know of any additional nondesktop usage criteria you'd like to see at this point? `days_seen_bits` can be used to calculate "Any Fennec Activity" as described in the [Engineering Design doc](https://docs.google.com/document/d/1L8tWDUjccutGGAldhpypRtPCaw3kkXboPUTtTZb02OA/edit#heading=h.sk8msaud272d) and I don't see any other nondesktop usage criteria listed there.

I'm also not going to include "seen_in_tier1_country" since I think we want to discourage using that except in the case of FxA where we're forced to.

Reply:
I'd like to have a separate usage criteria for each of the apps that exists in the table, as well as an "Any Non-desktop" criteria, although that could be something that just exists in the front-end if you prefer.  This should include activity from all Non-desktop products except Fenix (which uses Glean)?  Or am I misunderstanding?

Also, Would Fennec for iOS and Android be treated as separate apps?  Or do we have a OS slicing dimension?

Reply:
The current table here is built in top of core pings, so you are correct that Fenix and other new apps using Glean instead of core pings are not included at this point.

client_ids are separate between the different browser products, so if an Android user has Fennec and Focus installed, those two apps have different client_ids. Therefore, `days_seen_bits` really covers "Any nondesktop" and app_name is an exclusive dimension that we can filter on. So "Any nondesktop" with a filter of "app_name = Fennec AND os = Android" would give you "Any Fennec activity".

Integrating Glean is TBD. And I'm not sure what the plan is at this point about whether client_ids will be preserved in the Fennec to Fenix transition.